CMakeLists.txt

```diff
@@ -855,29 +761,156 @@ set(devilutionx_assets
   ui_art/mainmenuw.pcx
   ui_art/supportw.pcx)
 
+if(VIRTUAL_GAMEPAD)
+  list(APPEND devilutionx_assets
+    ui_art/button.png
+    ui_art/directions2.png
+    ui_art/directions.png
+    ui_art/menu-levelup.png
+    ui_art/menu.png)
+endif()
+
+if(NOT DEFINED DEVILUTIONX_ASSETS_OUTPUT_DIRECTORY)
+  set(DEVILUTIONX_ASSETS_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/assets")
+endif()
 
 # Copy assets to the build assets subdirectory. This serves two purposes:
 # - If smpq is installed, devilutionx.mpq is built from these files.
 # - If smpq is not installed, the game will load the assets directly from this directoy.
 foreach(asset_file ${devilutionx_assets})
   set(src "${CMAKE_CURRENT_SOURCE_DIR}/Packaging/resources/assets/${asset_file}")
   set(dst "${DEVILUTIONX_ASSETS_OUTPUT_DIRECTORY}/${asset_file}")
-  list(APPEND DEVILUTIONX_MPQ_FILES "${asset_file}")
   list(APPEND DEVILUTIONX_OUTPUT_ASSETS_FILES "${dst}")
+  list(APPEND DEVILUTIONX_MPQ_FILES "${asset_file}")
   add_custom_command(
     COMMENT "Copying ${asset_file}"
     OUTPUT "${dst}"
     DEPENDS "${src}"
     COMMAND ${CMAKE_COMMAND} -E copy "${src}" "${dst}"
     VERBATIM)
+
+  if(APPLE)
+    list(APPEND BIN_TARGET_SRCS ${dst})
+    string(REGEX MATCH "(.*)\/" subpath ${asset_file})
+    set_source_files_properties(${dst} PROPERTIES
+      MACOSX_PACKAGE_LOCATION "Resources/${subpath}"
+      XCODE_EXPLICIT_FILE_TYPE compiled)
+  endif()
 endforeach()
+
+set(devilutionx_langs bg cs da de es fr hr it ja ko_KR pl pt_BR ro_RO ru uk sv zh_CN zh_TW)
+if(USE_GETTEXT_FROM_VCPKG)
+  # vcpkg doesn't add its own tools directory to the search path
+  list(APPEND Gettext_ROOT ${CMAKE_CURRENT_BINARY_DIR}/vcpkg_installed/${VCPKG_TARGET_TRIPLET}/tools/gettext/bin)
+endif()
+find_package(Gettext)
 if (Gettext_FOUND)
   foreach(lang ${devilutionx_langs})
     list(APPEND DEVILUTIONX_MPQ_FILES "${lang}.gmo")
+
+	if(APPLE)
+      set(dst "${DEVILUTIONX_ASSETS_OUTPUT_DIRECTORY}/${lang}.gmo")
+      file(TOUCH ${dst})

```

Comment:
Whoa. I think we can simply add a dependency on lang targets to `BIN_TARGET` for mac/ios

```cmake
add_dependency(${BIN_TARGET} ${devilutionx_lang_targets} ${devilutionx_lang_files})
```

Reply:
I was unable to get it working with out adding them to the exec, but couldn't do so with out them existing at config time. I do realize that this is terrible solution :)